src/s2e/Plugins/VulnerabilityAnalysis/PovGenerator.cpp

```diff
@@ -50,38 +37,70 @@ std::string PovGenerator::writeToFile(S2EExecutionState *state, const PovOptions
             break;
     }
 
-    povFileNameSS << state->getID() << "." << fileExtWithoutDot;
-    if (m_compress) {
-        povFileNameSS << ".zlib"; // use `zlib-flate -uncompress`
+    povFileNameSS << state->getID();
+
+    if (fileExtWithoutDot.size() > 0) {
+        povFileNameSS << "." << fileExtWithoutDot;
     }
 
-    std::string povFileName = g_s2e->getOutputFilename(povFileNameSS.str());
+    return g_s2e->getOutputFilename(povFileNameSS.str());
```

Comment:
Can we use `s2e()->` instead? Using global vars like that is a little gross :)

Reply:
yes

Reply:
actually, no. These are all static functions, they don't have s2e() there.